playground/src/components/state-machines-details/state-machines-details.js

```diff
@@ -15,84 +15,116 @@ import {
+    this.fetchData(this.props.id);
+  }
+  componentDidUpdate(prevProps) {
+    if (prevProps.id !== this.props.id) {
+      this.fetchData(this.props.id);
+    } else if (prevProps.dataLocale !== this.props.dataLocale) {
+      this.fetchData(this.props.id);
+    }
+  }
+  componentWillUnmount() {
+    this.isUnmounting = true;
+  }
+  fetchData = id => {
+    this.setState({ isLoading: true, data: null, error: null });
+    this.props.dispatch(fetchStateMachine(id)).then(
       data => {
-        this.setState({ data, isLoading: false });
+        !this.isUnmounting &&
+          this.setState({ isLoading: false, data, error: null });
       },
       error => {
-        this.setState({ error, isLoading: false });
-        throw error;
+        !this.isUnmounting &&
+          this.setState({ isLoading: false, data: null, error });
+        this.props.showApiErrorNotification({ errors: [error] });
       }
     );
-  }
-
+  };
   render() {
-    if (this.state.isLoading) {
-      return <LoadingSpinner />;
-    }
-    return (
-      <Spacings.Inset scale="m">
-        <Spacings.Stack scale="l">
-          <LinkButton
-            iconLeft={<ListIcon />}
-            label="Back to list"
-            to={this.props.backToListPath}
-          />
-          <Spacings.Stack scale="xs">
-            <Text.Headline as="h2">
-              {this.state.data.name
-                ? this.state.data.name[this.props.dataLocale]
-                : 'n/a'}
-            </Text.Headline>
-            <Text.Detail>{this.state.data.key}</Text.Detail>
-          </Spacings.Stack>
-          <Constraints.Horizontal constraint="m">
-            <Grid
-              gridGap="16px"
-              gridAutoColumns="1fr"
-              gridTemplateColumns="repeat(2, 1fr)"
-            >
-              <Text.Body>{'Type'}</Text.Body>
-              <Text.Body>{this.state.data.type}</Text.Body>
-              <Text.Body>{'Built In'}</Text.Body>
-              <Text.Body>
-                {this.state.data.builtIn ? <CheckBoldIcon /> : ''}
-              </Text.Body>
-              <Text.Body>{'Initial'}</Text.Body>
-              <Text.Body>
-                {this.state.data.initial ? <CheckBoldIcon /> : ''}
-              </Text.Body>
-            </Grid>
-          </Constraints.Horizontal>
-        </Spacings.Stack>
-      </Spacings.Inset>
-    );
+    return this.props.children(this.state);
   }
 }
 
-export default flowRight(
-  withApplicationContext(context => ({ dataLocale: context.dataLocale })),
-  connect(
-    null,
-    {
-      fetchStateMachine,
-    }
-  )
-)(StateMachinesDetails);
+const StateMachinesDetails = props => {
+  const dataLocale = useApplicationContext(context => context.dataLocale);
+  const dispatch = useDispatch();
+  const showApiErrorNotification = useShowApiErrorNotification();
+
+  return (
+    <StateMachinesDetailsFetcher
+      id={props.id}
+      dataLocale={dataLocale}
+      dispatch={dispatch}
```

Comment:
Ah, can we not separate concerns here and make the child less aware and not pass `dispatch` but a bound action creator already?

Reply:
See https://github.com/commercetools/merchant-center-application-kit/pull/856#discussion_r303628451

Reply:
That part is clear. I was just wondering is we can not pass disptach down but bound action creators.

Reply:
Changed, thanks